core/src/main/java/org/springframework/security/oauth2/server/authorization/web/OAuth2AuthorizationEndpointFilter.java

```diff
@@ -15,33 +15,252 @@
+
+		return client;
+
+	}
+
+	protected OAuth2Authorization buildOAuth2Authorization(RegisteredClient client,
+			OAuth2AuthorizationRequest authorizationRequest, String code) {
+		OAuth2Authorization authorization = OAuth2Authorization.createBuilder()
+					.clientId(authorizationRequest.getClientId())
+					.addAttribute(OAuth2ParameterNames.CODE, code)
+					.addAttribute(OAuth2Authorization.ISSUED_AT, Instant.now())
+					.addAttribute(OAuth2Authorization.CODE_USED, new AtomicBoolean(false))
+					.addAttribute(OAuth2ParameterNames.SCOPE, Optional.ofNullable(authorizationRequest.getScopes())
+							.filter(scopes -> !scopes.isEmpty()).orElse(client.getScopes()))
+					.build();
+
+		return authorization;
+	}
+
+
+	protected void validateAuthorizationRequest(HttpServletRequest request, RegisteredClient client) {
+		String responseType = request.getParameter(OAuth2ParameterNames.RESPONSE_TYPE);
+		if (StringUtils.isEmpty(responseType)
+				|| !responseType.equals(OAuth2AuthorizationResponseType.CODE.getValue()))
+			throw new OAuth2AuthorizationException(RESPONSE_TYPE_NOT_FOUND_ERROR);
+
+		String redirectUri = request.getParameter(OAuth2ParameterNames.REDIRECT_URI);
+		if (StringUtils.isEmpty(redirectUri) && client.getRedirectUris().size() > 1)
+			throw new OAuth2AuthorizationException(REDIRECT_URI_REQUIRED);
+		if (!StringUtils.isEmpty(redirectUri) && !client.getRedirectUris().contains(redirectUri))
+			throw new OAuth2AuthorizationException(INVALID_REDIRECT_URI_REQUESTED);
+	}
+
+	private String getRedirectUri(OAuth2AuthorizationRequest authorizationRequest, RegisteredClient client) {
+		return !StringUtils.isEmpty(authorizationRequest.getRedirectUri())
+		? authorizationRequest.getRedirectUri()
+		: client.getRedirectUris().stream().findFirst().get();
+	}
+
+	private void sendCodeOnSuccess(HttpServletRequest request, HttpServletResponse response,
+			OAuth2AuthorizationRequest authorizationRequest, String redirectUri, String code) throws IOException {
+		UriComponentsBuilder redirectUriBuilder = UriComponentsBuilder.fromUriString(redirectUri)
+				.queryParam(OAuth2ParameterNames.CODE, code);
+		if (!StringUtils.isEmpty(authorizationRequest.getState()))
+			redirectUriBuilder.queryParam(OAuth2ParameterNames.STATE, authorizationRequest.getState());
+
+		String finalRedirectUri = redirectUriBuilder.toUriString();
+		this.authorizationRedirectStrategy.sendRedirect(request, response, finalRedirectUri);
+	}
+
+	private void sendErrorInResponse(HttpServletResponse response, OAuth2Error authorizationError) throws IOException {
+		int errorStatus = -1;
+		String errorCode = authorizationError.getErrorCode();
+		if (errorCode.equals(OAuth2ErrorCodes.ACCESS_DENIED))
+			errorStatus=HttpStatus.FORBIDDEN.value();
+		else errorStatus=HttpStatus.INTERNAL_SERVER_ERROR.value();
+		response.sendError(errorStatus, authorizationError.getErrorCode()+":"+authorizationError.getDescription());
```

Comment:
I find not using curly braces in "if" statements to be very bug prone. I'd suggest introducing them here and in other other locations in this PR. I've noticed this in numerous locations within this PR

Reply:
@jgrandja do you have any suggestions on which coding guidelines we should follow? I found [this document](https://github.com/spring-projects/spring-framework/wiki/Code-Style) in the spring-framework repository. But it does say its still a work-in-progress

Reply:
@pgerhard Take a look at the [Getting Started](https://github.com/spring-projects-experimental/spring-authorization-server#getting-started), specifically:

> Submitted work via pull requests should follow the same coding style/conventions and adopt the same or similar design patterns that have been established in Spring Security’s OAuth 2.0 support

Essentially, we are looking for the same coding style/patterns that are implemented in the OAuth 2.0 codebase in Spring Security 5.x. Consistency is key, so the Authorization Server code should align as well.